tests/unit/module_utils/test_cloud.py

```diff
@@ -0,0 +1,200 @@
+# (c) 2021 Red Hat Inc.
+#
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible_collections.amazon.aws.plugins.module_utils.cloud import CloudRetry, BackoffIterator
+import pytest
+import unittest
+import random
+from datetime import datetime
+
+
+def test_backoff_value_generator():
+    max_delay = 60
+    initial = 3
+    backoff = 2
+
+    min_sleep = initial
+    counter = 0
+    for sleep in BackoffIterator(delay=initial, backoff=backoff, max_delay=max_delay):
+        if counter > 4:
+            assert sleep == max_delay
+        else:
+            assert sleep == min_sleep
+            min_sleep *= backoff
+        counter += 1
+        if counter == 10:
+            break
+
+
+def test_backoff_value_generator_with_jitter():
+    max_delay = 60
+    initial = 3
+    backoff = 2
+
+    min_sleep = initial
+    counter = 0
+    for sleep in BackoffIterator(delay=initial, backoff=backoff, max_delay=max_delay, jitter=True):
+        if counter > 4:
+            assert sleep <= max_delay
+        else:
+            assert sleep <= min_sleep
+            min_sleep *= backoff
+        counter += 1
+        if counter == 10:
+            break
+
+
+class CloudRetryUtils(unittest.TestCase):
```

Comment:
I don't know for use, but ansible/ansible is moving toward pytest as the default. Please don't rewrite your tests thus :-).

Reply:
I just created something similar to what is already done in this repository, but maybe this can be changed

Reply:
The structure used by @abikouo is the same as the one in the latest dev guide.  If this is incorrect, then the dev guide should be updated.

https://docs.ansible.com/ansible/latest/dev_guide/testing_units_modules.html#a-complete-example

Reply:
Oh indeed, there is example for both in this document, pytest and unittest.TestCase. It's a bit confusing.